File: frontend/src/lib/api/types/release.ts
/**
 * Release DTO shapes matching sprint-service `/api/releases`.
 */

export interface ReleaseDto {
  id: string;
  projectId: string;
  organizationId?: string | null;
  name: string;
  version: string | null;
  description: string | null;
  status: string;
  releaseDate: string | null;
  features: string[] | null;
  createdAt: string;
  updatedAt: string;
}

export interface CreateReleaseRequest {
  projectId: string;
  name: string;
  version?: string | null;
  description?: string | null;
  status: string;
  releaseDate?: string | null;
  features?: string[];
}

export interface UpdateReleaseRequest {
  name?: string;
  version?: string | null;
  description?: string | null;
  status?: string;
  releaseDate?: string | null;
  features?: string[];
}

export interface ReleaseListQuery {
  projectId?: string;
}

File: backend/services/sprint-service/src/main/java/com/devflow/sprint/entity/Release.java
package com.devflow.sprint.entity;

import com.devflow.common.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;

import java.time.LocalDate;
import java.util.UUID;

/**
 * A project release/version, tracked independently of any one sprint. A {@link Sprint} may
 * optionally reference one via {@code Sprint.releaseId}.
 */
@Entity
@Table(name = "releases")
public class Release extends BaseEntity {

    @Column(name = "project_id", nullable = false)
    private UUID projectId;

    @Column(name = "organization_id")
    private UUID organizationId;

    @Column(nullable = false, length = 160)
    private String name;

    @Column(length = 64)
    private String version;

    @Column(length = 2000)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 32)
    private ReleaseStatus status;

    @Column(name = "release_date")
    private LocalDate releaseDate;

    @Column(name = "features_json", columnDefinition = "TEXT")
    private String featuresJson;

    public UUID getProjectId() {
        return projectId;
    }

    public void setProjectId(UUID projectId) {
        this.projectId = projectId;
    }

    public UUID getOrganizationId() {
        return organizationId;
    }

    public void setOrganizationId(UUID organizationId) {
        this.organizationId = organizationId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getVersion() {
        return version;
    }

    public void setVersion(String version) {
        this.version = version;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public ReleaseStatus getStatus() {
        return status;
    }

    public void setStatus(ReleaseStatus status) {
        this.status = status;
    }

    public LocalDate getReleaseDate() {
        return releaseDate;
    }

    public void setReleaseDate(LocalDate releaseDate) {
        this.releaseDate = releaseDate;
    }

    public String getFeaturesJson() {
        return featuresJson;
    }

    public void setFeaturesJson(String featuresJson) {
        this.featuresJson = featuresJson;
    }
}
